### PR TITLE
Run tests in CI for aarch64-apple-darwin

### PR DESCRIPTION
**Pull request by shepmaster**
r? @ghost

### CODE REVIEW

**Comment by shepmaster** [comment]
This should be ready to go now!

r? @Mark-Simulacrum

File: tests/rustdoc-ui/issues/issue-107918.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // aux-build:panic-handler.rs
 // compile-flags: --document-private-items
 // build-pass
+// ignore-macos
```
Comment (GuillaumeGomez) [inline tests/rustdoc-ui/issues/issue-107918.rs:4]: Wouldn't it be simpler to just run it on linux instead?

**Comment by GuillaumeGomez** [comment]
> /cc @GuillaumeGomez as I'm touching the test you originally added.

Thanks, much appreciated! :)

File: tests/rustdoc-ui/issues/issue-107918.rs (hunk 1 above)
Comment (shepmaster) [reply tests/rustdoc-ui/issues/issue-107918.rs:4]: Simpler as in fewer lines, yes.

However, I don't have the context of the original point of the test so I can't speak to if the test is *supposed* to only be run on Linux. `ONLY Linux` and `NOT (macOS OR Windows)` produce different sets of platforms.

Comment (GuillaumeGomez) [reply tests/rustdoc-ui/issues/issue-107918.rs:4]: > Simpler as in fewer lines, yes.

More like "we don't want to check that each platform doesn't break".

> However, I don't have the context of the original point of the test so I can't speak to if the test is _supposed_ to only be run on Linux. `ONLY Linux` and `NOT (macOS OR Windows)` produce different sets of platforms.

The point of this test is to be a regression test for the #107918 issue. So as long as it's working on one platform, we're achieving this goal. However to do so, we use very uncommon code, so I think it's better to reduce the number of targets on which we actually check it for simpler maintenance.

So yes, please change it into only linux. :)

Comment (shepmaster) [reply tests/rustdoc-ui/issues/issue-107918.rs]: Changed!

Comment (GuillaumeGomez) [reply tests/rustdoc-ui/issues/issue-107918.rs]: Thanks!

**Review by GuillaumeGomez — APPROVED** [review]
rustdoc test changes look good to me!

**Comment by GuillaumeGomez** [comment]
Let's set a reviewer from the infra team now.

r? @onur-ozkan

File: src/ci/scripts/install-tidy.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# This script downloads and installs the tidy binary from Homebrew.
+
+set -euo pipefail
+IFS=$'\n\t'
+
+source "$(cd "$(dirname "$0")" && pwd)/../shared.sh"
+
+# Only the macOS arm64/aarch64 GitHub Actions runner needs to have tidy
+# installed; other platforms have it preinstalled.
```
Comment (onur-ozkan) [inline src/ci/scripts/install-tidy.sh:10]: Q: I wonder if we can install this on the image build directly.

**Review by onur-ozkan — APPROVED** [review]
Changes looks good to me other than 1 question

File: src/ci/scripts/install-tidy.sh (hunk 2 above)
Comment (shepmaster) [reply src/ci/scripts/install-tidy.sh:10]: [Not yet](https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/Github.20M1.20CI.20seems.20to.20exist.20now/near/394881350)

**Comment by onur-ozkan** [comment]
I don't know when you want to ship this so I leave it to you. Feel free to r=GuillaumeGomez,me.

**Comment by lqd** [comment]
The timeouts are most likely related to the perf collector, not this PR.

**Pull request merged**